compiler/src/iree/compiler/Codegen/Common/GPUVectorization.cpp

```diff
@@ -38,8 +42,28 @@ static void populateVectorizationPatterns(RewritePatternSet &patterns) {
        StringAttr::get(ctx, getVectorizeMarker())},
       llvm::None);
   f.setMatchByDefault();
+  // When vectorizing if some ops didn't get tiled we may end up with large
+  // vectors being created that will later explode code size. If we have any
+  // vectors larger that what would fit in register skip vectorization.
+  f.addFilter([](Operation *op) {
+    auto linalgOp = dyn_cast<linalg::LinalgOp>(op);
+    if (!linalgOp) return success();
+    int64_t maxFlatVecSize = 1;
+    for (OpOperand *operand : linalgOp.getInputAndOutputOperands()) {
+      auto type = operand->get().getType().dyn_cast<ShapedType>();
+      if (!type) continue;
+      int64_t flatVecSize = 1;
+      for (int64_t dim : type.getShape()) {
+        if (ShapedType::isDynamic(dim)) return failure();
+        flatVecSize *= dim;
```

Comment:
We can just call`getNumElements` after checking the shape is static?

Reply:
yes this is simpler indeed.